client/src/components/TagSelector/TagSelector.style.ts

```diff
@@ -94,15 +95,214 @@ export const Title = styled.h3`
+	height: 20px;
+
+	&:hover {
+		background-color: deepskyblue;
+		color: white;
+		border-radius: 50%;
+	}
 `;
+
+ClearFilter.defaultProps = {
+	type: "button",
+};
+
+const FilterWrapper = styled.div`
+	position: relative;
+	display: flex;
+	flex-direction: row;
+	gap: 0;
+	align-items: center;
+	justify-content: center;
+`;
+
+const Actions = styled.div`
+	display: flex;
+	flex-direction: row;
+	justify-content: space-between;
+	align-items: center;
+	margin-top: 0.3rem;
+	gap: 0.5rem;
+
+	button:nth-of-type(1) {
+		margin-left: auto;
+	}
+`;
+
+const Dropdown = styled.div`
+	position: relative;
+`;
+
+const DropdownTrigger = styled.button`
+	border-radius: 50%;
+	width: 30px;
+	height: 30px;
+	display: flex;
+	justify-content: center;
+	align-items: center;
+	padding: 0px;
+	outline: none;
+	border: none;
+
+	&:active {
+		transform: scale(1.1);
+		background-color: white;
+		& > svg {
+			fill: black;
+		}
+	}
+
+	&:focus:not(:active) {
+		outline: 2px solid dodgerblue;
+		background-color: #fff;
+	}
+
+	&:hover {
+		background-color: #fafafa;
+		box-shadow: 0 0.2rem 0.5rem 0 #999;
+	}
+`;
+
+DropdownTrigger.defaultProps = {
+	type: "button",
+};
+
+const DropdownActions = styled.div`
+	display: flex;
+	flex-direction: row;
+	gap: 1rem;
+	align-items: flex-end;
+	margin: 0.8rem 1rem;
+	justify-content: space-between;
+
+	button:nth-of-type(1) {
+		margin-left: auto;
+	}
+
+	${Actions} {
+		justify-content: flex-start;
+		align-items: flex-start;
+
+		& > ${Filter} {
+			align-self: flex-start;
+		}
+	}
+`;
+
+const DropdownContent = styled.div`
+	position: absolute;
+	top: -0.8rem;
+	left: -1.1rem;
+	display: flex;
+	flex-direction: column;
+	background-color: #eee;
+	border: 2px solid #ddd;
+	border-radius: 4px;
+	box-shadow:
+		0.2rem 0.2rem 0 0 #333,
+		0 0 0.6rem 0 #999;
+	width: calc(
+		100% + 2.2rem
+	); // 100% plus twice the left offset to center it against its parent
+`;
+
+const SelectionList = styled.ul`
+	padding: 0.2rem 0;
+	margin: 0.7rem 0.4rem;
+	display: flex;
+	padding-right: 0.4rem; // this is to prevent make scrollbar look better
+	flex-direction: row;
+	flex-wrap: wrap;
+	gap: 0.5rem;
+	font-size: 0.85rem;
+	align-items: center;
+	user-select: none;
+	max-height: 120px;
+	overflow-y: scroll;
+`;
+
+const SelectionItem = styled.li`
+	list-style: none;
+	display: flex;
+	padding: 0.3rem 0.5rem;
+	min-width: max-content;
+	flex: 1;
+	border-radius: 4px;
+	background-color: dodgerblue;
+	color: white;
+	justify-content: center;
+	align-items: center;
+	box-shadow: 0rem 0.2rem 0.2rem #ccc;
+`;
+
+const PathPart = styled.span<{ $isLeaf: boolean }>`
+	color: ${(p) => (p.$isLeaf ? "white" : "lightblue")};
+	${(p) =>
+		!p.$isLeaf &&
+		css`
+			max-width: 75px;
+			overflow: hidden;
+			text-overflow: ellipsis;
+			white-space: nowrap;
+		`}
+`;
+
+const EmptySelection = styled.div`
+	padding: 0.4rem 1.2rem;
+	color: azure;
+	background-color: dodgerblue;
+	max-width: max-content;
+	margin-top: 0.5rem;
+`;
+
+export default {
```

Comment:
For this one, I put all these in the default export instead of exporting all styled components invidually by name. I think I prefer things this way, so it's harder to accidentally import something in the wrong place, and to keep things more confined. Should do a refactor pass later on to ensure consistency here.

Reply:
I implemented an eslint rule to enforce this, but since we're not running any checks on commit/push it's not really enforced yet. Squiggly lines are probably incentive enough for me to fix it soon though.